src/types/resolveConstantValue.ts

```diff
@@ -59,6 +59,18 @@ function reduceInt(ast: ASTExpression): bigint {
     throwError('Cannot reduce expression to a constant integer', ast.ref);
 }
 
+function reduceInt(ast: ASTExpression): bigint {
+  try {
+    return reduceIntImpl(ast)
+  } catch (error) {
+    if (error instanceof RangeError) {
+      throwError('Cannot evaluate expression due to integer overflow', ast.ref);
```

Comment:
maybe something like this?
```suggestion
      throwError('Cannot evaluate constant expression due to integer overflow', ast.ref);
```